app/core/AppStateManager.test.ts

```diff
@@ -0,0 +1,152 @@
+import { AppState, AppStateStatus } from 'react-native';
+import { MetaMetrics, MetaMetricsEvents } from './Analytics';
+import { store } from '../store';
+import AppStateManager from './AppStateManager';
+import extractURLParams from './DeeplinkManager/ParseManager/extractURLParams';
+import Logger from '../util/Logger';
+
+jest.mock('react-native', () => ({
+  AppState: {
+    addEventListener: jest.fn(),
+    currentState: 'active',
+  },
+}));
+
+jest.mock('./Analytics', () => ({
+  MetaMetrics: {
+    getInstance: jest.fn(),
+  },
+  MetaMetricsEvents: {
+    APP_OPENED: 'APP_OPENED',
+  },
+}));
+
+jest.mock('../store', () => ({
+  store: {
+    getState: jest.fn(),
+  },
+}));
+
+jest.mock('./DeeplinkManager/ParseManager/extractURLParams', () => jest.fn());
+
+jest.mock('../util/Logger', () => ({
+  error: jest.fn(),
+}));
+
+describe('AppStateManager', () => {
+  let appStateManager: AppStateManager;
+  let mockAppStateListener: (state: AppStateStatus) => void;
+  let mockTrackEvent: jest.Mock;
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+    jest.useFakeTimers();
+    mockTrackEvent = jest.fn();
+    (MetaMetrics.getInstance as jest.Mock).mockReturnValue({
+      trackEvent: mockTrackEvent,
+    });
+    (AppState.addEventListener as jest.Mock).mockImplementation((_, listener) => {
+      mockAppStateListener = listener;
+      return { remove: jest.fn() };
+    });
+    appStateManager = new AppStateManager();
+  });
+
+  afterEach(() => {
+    jest.useRealTimers();
+  });
+
+  it('should subscribe to AppState changes on instantiation', () => {
+    expect(AppState.addEventListener).toHaveBeenCalledWith('change', expect.any(Function));
+  });
+
+  it('should track event when app becomes active and conditions are met', () => {
+    (store.getState as jest.Mock).mockReturnValue({
+      security: { dataCollectionForMarketing: true },
+    });
+    (extractURLParams as jest.Mock).mockReturnValue({ params: { attributionId: 'test123' } });
+
+    appStateManager.setCurrentDeeplink('metamask://connect?attributionId=test123');
+    mockAppStateListener('active');
+    jest.advanceTimersByTime(2000);
+
+    expect(mockTrackEvent).toHaveBeenCalledWith(
+      MetaMetricsEvents.APP_OPENED,
+      { attributionId: 'test123' },
+      true
+    );
+  });
+
+  it('should not track event when data collection is disabled', () => {
+    (store.getState as jest.Mock).mockReturnValue({
+      security: { dataCollectionForMarketing: false },
+    });
+
+    mockAppStateListener('active');
+    jest.advanceTimersByTime(2000);
+
+    expect(mockTrackEvent).toHaveBeenCalledWith(
+      MetaMetricsEvents.APP_OPENED,
+      { attributionId: undefined },
+      true
+    );
+  });
+
+  it('should not track event when there is no deeplink', () => {
```

Comment:
Q: what does not tracking mean here? I can read the assert toHaveBeenCalledWith so it means trackEvent is called.

Also naming guidelines:
```suggestion
  it('does not track event when there is no deeplink', () => {
```